spec/models/king_spec.rb

```diff
@@ -1,95 +1,96 @@
 require 'rails_helper'
 
 RSpec.describe King, type: :model do
-  describe 'white king movement validation' do
-    it 'should return false if not moved' do
-      king = FactoryGirl.create(:king, :white)
-      expect(king.valid_move?(3, 7)).to eq false
-    end
-
-    it 'should return false if moved 2 spaces right' do
-      king = FactoryGirl.create(:king, :white)
-      expect(king.valid_move?(5, 7)).to eq false
-    end
-
-    it 'should return false if moved 3 spaces forward and 3 spaces right' do
-      king = FactoryGirl.create(:king, :white)
-      expect(king.valid_move?(6, 4)).to eq false
-    end
-
-    it 'should return true if moved 1 space forward and 1 space right' do
-      king = FactoryGirl.create(:king, :white)
-      expect(king.valid_move?(4, 6)).to eq true
-    end
+  let(:game) { FactoryGirl.create(:game) }
+  let(:king) do
+    game.pieces.find_by(
+      type: 'King',
+      color: 'white',
+      x_position: 4,
+      y_position: 7
+    )
+  end
+  let(:moved_king) do
+    game.pieces.create(
+      type: 'King',
+      color: 'white',
+      x_position: 3,
+      y_position: 3,
+      moved: true
+    )
+  end
 
-    it 'should return true if moved to 1 space left' do
-      king = FactoryGirl.create(:king, :white)
-      expect(king.valid_move?(2, 7)).to eq true
+  describe 'creation' do
+    it 'should create a white king' do
+      king = FactoryGirl.create(:king, color: 'white')
+      expect(king.type).to eq('King')
     end
 
-    it 'should return true if moved 1 space forward' do
-      king = FactoryGirl.create(:king, :white)
-      expect(king.valid_move?(3, 6)).to eq true
+    it 'should fail to create a red king' do
+      expect { FactoryGirl.create(:king, color: 'red') }.to\
+        raise_error(ActiveRecord::RecordInvalid)
     end
   end
 
-  describe 'black king movement validation' do
+  describe 'moved' do
     it 'should return false if not moved' do
-      king = FactoryGirl.create(:king, :black)
-      expect(king.valid_move?(3, 0)).to eq false
-    end
-
-    it 'should return false if moved 2 spaces right' do
-      king = FactoryGirl.create(:king, :black)
-      expect(king.valid_move?(5, 0)).to eq false
-    end
-
-    it 'should return false if moved 3 spaces forward and 3 spaces right' do
-      king = FactoryGirl.create(:king, :black)
-      expect(king.valid_move?(6, 3)).to eq false
+      expect(king.move!(4, 7)).to eq false
+      expect(king.x_position).to eq 4
+      expect(king.y_position).to eq 7
+      expect(king.moved).to eq false
     end
+  end
 
-    it 'should return true if moved 1 space forward and 1 space right' do
-      king = FactoryGirl.create(:king, :black)
-      expect(king.valid_move?(4, 1)).to eq true
+  describe 'invalid moveset' do
```

Comment:
See above, code is here:

`describe 'moved' do
    it 'should return false if not moved' do
      expect(king.move!(4, 7)).to eq false
      expect(king.x_position).to eq 4
      expect(king.y_position).to eq 7
      expect(king.moved).to eq false
    end
  end`